### PR TITLE
`LazyBuildMixInTest#newRunningBuildRelationFromPrevious` flaked on Windows

### PR DESCRIPTION
**Pull request by basil**
Observed in #6173. This PR fixes the test by ensuring that we wait for the build to complete before tearing down Jenkins.

<details>
<summary>Stack trace</summary>
<pre>
jenkins.util.io.CompositeIOException: Unable to delete 'C:\Jenkins\workspace\Core_jenkins_PR-6173\test\target\j h11828234848007349220'. Tried 3 times (of a maximum of 3) waiting 0.1 sec between attempts.
	at jenkins.util.io.PathRemover.forceRemoveRecursive(PathRemover.java:96)
	at hudson.Util.deleteRecursive(Util.java:319)
	at hudson.FilePath$DeleteRecursive.invoke(FilePath.java:1444)
	at hudson.FilePath$DeleteRecursive.invoke(FilePath.java:1439)
	at hudson.FilePath.act(FilePath.java:1200)
	at hudson.FilePath.act(FilePath.java:1183)
	at hudson.FilePath.deleteRecursive(FilePath.java:1436)
	at org.jvnet.hudson.test.TemporaryDirectoryAllocator.dispose(TemporaryDirectoryAllocator.java:93)
	at org.jvnet.hudson.test.TestEnvironment.dispose(TestEnvironment.java:84)
	at org.jvnet.hudson.test.JenkinsRule.after(JenkinsRule.java:520)
	at org.jvnet.hudson.test.JenkinsRule$1.evaluate(JenkinsRule.java:619)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:299)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:293)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.lang.Thread.run(Thread.java:829)
	Suppressed: jenkins.util.io.CompositeIOException: Unable to remove directory C:\Jenkins\workspace\Core_jenkins_PR-6173\test\target\j h11828234848007349220\jobs\test0\builds\2 with directory contents: [C:\Jenkins\workspace\Core_jenkins_PR-6173\test\target\j h11828234848007349220\jobs\test0\builds\2\atomic3310657271750370634tmp]
		at jenkins.util.io.PathRemover.removeOrMakeRemovableThenRemove(PathRemover.java:249)
		at jenkins.util.io.PathRemover.tryRemoveFile(PathRemover.java:202)
		at jenkins.util.io.PathRemover.tryRemoveRecursive(PathRemover.java:213)
		at jenkins.util.io.PathRemover.tryRemoveDirectoryContents(PathRemover.java:224)
		at jenkins.util.io.PathRemover.tryRemoveRecursive(PathRemover.java:212)
		at jenkins.util.io.PathRemover.tryRemoveDirectoryContents(PathRemover.java:224)
		at jenkins.util.io.PathRemover.tryRemoveRecursive(PathRemover.java:212)
		at jenkins.util.io.PathRemover.tryRemoveDirectoryContents(PathRemover.java:224)
		at jenkins.util.io.PathRemover.tryRemoveRecursive(PathRemover.java:212)
		at jenkins.util.io.PathRemover.tryRemoveDirectoryContents(PathRemover.java:224)
		at jenkins.util.io.PathRemover.tryRemoveRecursive(PathRemover.java:212)
		at jenkins.util.io.PathRemover.forceRemoveRecursive(PathRemover.java:93)
		... 14 more
		Suppressed: java.nio.file.DirectoryNotEmptyException: C:\Jenkins\workspace\Core_jenkins_PR-6173\test\target\j h11828234848007349220\jobs\test0\builds\2
			at java.base/sun.nio.fs.WindowsFileSystemProvider.implDelete(WindowsFileSystemProvider.java:271)
			at java.base/sun.nio.fs.AbstractFileSystemProvider.deleteIfExists(AbstractFileSystemProvider.java:110)
			at java.base/java.nio.file.Files.deleteIfExists(Files.java:1181)
			at jenkins.util.io.PathRemover.removeOrMakeRemovableThenRemove(PathRemover.java:236)
			... 25 more
		Suppressed: java.nio.file.DirectoryNotEmptyException: C:\Jenkins\workspace\Core_jenkins_PR-6173\test\target\j h11828234848007349220\jobs\test0\builds\2
			at java.base/sun.nio.fs.WindowsFileSystemProvider.implDelete(WindowsFileSystemProvider.java:271)
			at java.base/sun.nio.fs.AbstractFileSystemProvider.deleteIfExists(AbstractFileSystemProvider.java:110)
			at java.base/java.nio.file.Files.deleteIfExists(Files.java:1181)
			at jenkins.util.io.PathRemover.removeOrMakeRemovableThenRemove(PathRemover.java:240)
			... 25 more
	Suppressed: jenkins.util.io.CompositeIOException: Unable to remove directory C:\Jenkins\workspace\Core_jenkins_PR-6173\test\target\j h11828234848007349220\jobs\test0\builds with directory contents: [C:\Jenkins\workspace\Core_jenkins_PR-6173\test\target\j h11828234848007349220\jobs\test0\builds\2]
		at jenkins.util.io.PathRemover.removeOrMakeRemovableThenRemove(PathRemover.java:249)
		at jenkins.util.io.PathRemover.tryRemoveFile(PathRemover.java:202)
		at jenkins.util.io.PathRemover.tryRemoveRecursive(PathRemover.java:213)
		at jenkins.util.io.PathRemover.tryRemoveDirectoryContents(PathRemover.java:224)
		at jenkins.util.io.PathRemover.tryRemoveRecursive(PathRemover.java:212)
		at jenkins.util.io.PathRemover.tryRemoveDirectoryContents(PathRemover.java:224)
		at jenkins.util.io.PathRemover.tryRemoveRecursive(PathRemover.java:212)
		at jenkins.util.io.PathRemover.tryRemoveDirectoryContents(PathRemover.java:224)
		at jenkins.util.io.PathRemover.tryRemoveRecursive(PathRemover.java:212)
		at jenkins.util.io.PathRemover.forceRemoveRecursive(PathRemover.java:93)
		... 14 more
		Suppressed: java.nio.file.DirectoryNotEmptyException: C:\Jenkins\workspace\Core_jenkins_PR-6173\test\target\j h11828234848007349220\jobs\test0\builds
			at java.base/sun.nio.fs.WindowsFileSystemProvider.implDelete(WindowsFileSystemProvider.java:271)
			at java.base/sun.nio.fs.AbstractFileSystemProvider.deleteIfExists(AbstractFileSystemProvider.java:110)
			at java.base/java.nio.file.Files.deleteIfExists(Files.java:1181)
			at jenkins.util.io.PathRemover.removeOrMakeRemovableThenRemove(PathRemover.java:236)
			... 23 more
		Suppressed: java.nio.file.DirectoryNotEmptyException: C:\Jenkins\workspace\Core_jenkins_PR-6173\test\target\j h11828234848007349220\jobs\test0\builds
			at java.base/sun.nio.fs.WindowsFileSystemProvider.implDelete(WindowsFileSystemProvider.java:271)
			at java.base/sun.nio.fs.AbstractFileSystemProvider.deleteIfExists(AbstractFileSystemProvider.java:110)
			at java.base/java.nio.file.Files.deleteIfExists(Files.java:1181)
			at jenkins.util.io.PathRemover.removeOrMakeRemovableThenRemove(PathRemover.java:240)
			... 23 more
	Suppressed: jenkins.util.io.CompositeIOException: Unable to remove directory C:\Jenkins\workspace\Core_jenkins_PR-6173\test\target\j h11828234848007349220\jobs\test0 with directory contents: [C:\Jenkins\workspace\Core_jenkins_PR-6173\test\target\j h11828234848007349220\jobs\test0\builds]
		at jenkins.util.io.PathRemover.removeOrMakeRemovableThenRemove(PathRemover.java:249)
		at jenkins.util.io.PathRemover.tryRemoveFile(PathRemover.java:202)
		at jenkins.util.io.PathRemover.tryRemoveRecursive(PathRemover.java:213)
		at jenkins.util.io.PathRemover.tryRemoveDirectoryContents(PathRemover.java:224)
		at jenkins.util.io.PathRemover.tryRemoveRecursive(PathRemover.java:212)
		at jenkins.util.io.PathRemover.tryRemoveDirectoryContents(PathRemover.java:224)
		at jenkins.util.io.PathRemover.tryRemoveRecursive(PathRemover.java:212)
		at jenkins.util.io.PathRemover.forceRemoveRecursive(PathRemover.java:93)
		... 14 more
		Suppressed: java.nio.file.DirectoryNotEmptyException: C:\Jenkins\workspace\Core_jenkins_PR-6173\test\target\j h11828234848007349220\jobs\test0
			at java.base/sun.nio.fs.WindowsFileSystemProvider.implDelete(WindowsFileSystemProvider.java:271)
			at java.base/sun.nio.fs.AbstractFileSystemProvider.deleteIfExists(AbstractFileSystemProvider.java:110)
			at java.base/java.nio.file.Files.deleteIfExists(Files.java:1181)
			at jenkins.util.io.PathRemover.removeOrMakeRemovableThenRemove(PathRemover.java:236)
			... 21 more
		Suppressed: java.nio.file.DirectoryNotEmptyException: C:\Jenkins\workspace\Core_jenkins_PR-6173\test\target\j h11828234848007349220\jobs\test0
			at java.base/sun.nio.fs.WindowsFileSystemProvider.implDelete(WindowsFileSystemProvider.java:271)
			at java.base/sun.nio.fs.AbstractFileSystemProvider.deleteIfExists(AbstractFileSystemProvider.java:110)
			at java.base/java.nio.file.Files.deleteIfExists(Files.java:1181)
			at jenkins.util.io.PathRemover.removeOrMakeRemovableThenRemove(PathRemover.java:240)
			... 21 more
	Suppressed: jenkins.util.io.CompositeIOException: Unable to remove directory C:\Jenkins\workspace\Core_jenkins_PR-6173\test\target\j h11828234848007349220\jobs with directory contents: [C:\Jenkins\workspace\Core_jenkins_PR-6173\test\target\j h11828234848007349220\jobs\test0]
		at jenkins.util.io.PathRemover.removeOrMakeRemovableThenRemove(PathRemover.java:249)
		at jenkins.util.io.PathRemover.tryRemoveFile(PathRemover.java:202)
		at jenkins.util.io.PathRemover.tryRemoveRecursive(PathRemover.java:213)
		at jenkins.util.io.PathRemover.tryRemoveDirectoryContents(PathRemover.java:224)
		at jenkins.util.io.PathRemover.tryRemoveRecursive(PathRemover.java:212)
		at jenkins.util.io.PathRemover.forceRemoveRecursive(PathRemover.java:93)
		... 14 more
		Suppressed: java.nio.file.DirectoryNotEmptyException: C:\Jenkins\workspace\Core_jenkins_PR-6173\test\target\j h11828234848007349220\jobs
			at java.base/sun.nio.fs.WindowsFileSystemProvider.implDelete(WindowsFileSystemProvider.java:271)
			at java.base/sun.nio.fs.AbstractFileSystemProvider.deleteIfExists(AbstractFileSystemProvider.java:110)
			at java.base/java.nio.file.Files.deleteIfExists(Files.java:1181)
			at jenkins.util.io.PathRemover.removeOrMakeRemovableThenRemove(PathRemover.java:236)
			... 19 more
		Suppressed: java.nio.file.DirectoryNotEmptyException: C:\Jenkins\workspace\Core_jenkins_PR-6173\test\target\j h11828234848007349220\jobs
			at java.base/sun.nio.fs.WindowsFileSystemProvider.implDelete(WindowsFileSystemProvider.java:271)
			at java.base/sun.nio.fs.AbstractFileSystemProvider.deleteIfExists(AbstractFileSystemProvider.java:110)
			at java.base/java.nio.file.Files.deleteIfExists(Files.java:1181)
			at jenkins.util.io.PathRemover.removeOrMakeRemovableThenRemove(PathRemover.java:240)
			... 19 more
	Suppressed: jenkins.util.io.CompositeIOException: Unable to remove directory C:\Jenkins\workspace\Core_jenkins_PR-6173\test\target\j h11828234848007349220 with directory contents: [C:\Jenkins\workspace\Core_jenkins_PR-6173\test\target\j h11828234848007349220\jobs]
		at jenkins.util.io.PathRemover.removeOrMakeRemovableThenRemove(PathRemover.java:249)
		at jenkins.util.io.PathRemover.tryRemoveFile(PathRemover.java:202)
		at jenkins.util.io.PathRemover.tryRemoveRecursive(PathRemover.java:213)
		at jenkins.util.io.PathRemover.forceRemoveRecursive(PathRemover.java:93)
		... 14 more
		Suppressed: java.nio.file.DirectoryNotEmptyException: C:\Jenkins\workspace\Core_jenkins_PR-6173\test\target\j h11828234848007349220
			at java.base/sun.nio.fs.WindowsFileSystemProvider.implDelete(WindowsFileSystemProvider.java:271)
			at java.base/sun.nio.fs.AbstractFileSystemProvider.deleteIfExists(AbstractFileSystemProvider.java:110)
			at java.base/java.nio.file.Files.deleteIfExists(Files.java:1181)
			at jenkins.util.io.PathRemover.removeOrMakeRemovableThenRemove(PathRemover.java:236)
			... 17 more
		Suppressed: java.nio.file.DirectoryNotEmptyException: C:\Jenkins\workspace\Core_jenkins_PR-6173\test\target\j h11828234848007349220
			at java.base/sun.nio.fs.WindowsFileSystemProvider.implDelete(WindowsFileSystemProvider.java:271)
			at java.base/sun.nio.fs.AbstractFileSystemProvider.deleteIfExists(AbstractFileSystemProvider.java:110)
			at java.base/java.nio.file.Files.deleteIfExists(Files.java:1181)
			at jenkins.util.io.PathRemover.removeOrMakeRemovableThenRemove(PathRemover.java:240)
			... 17 more
</pre>
</details>

### Proposed changelog entries

N/A

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [x] (If applicable) Jira issue is well described
- [x] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [x] Appropriate autotests or explanation to why this change has no tests
- [x] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

**Review by MarkEWaite — APPROVED** [review]

**Comment by basil** [comment]
This PR is now ready for merge. We will merge it after approximately 24 hours if there is no negative feedback. Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process. Thanks!

**Pull request merged**
